src/app/components/ui/projectForm/ProjectForm.tsx

```diff
@@ -0,0 +1,105 @@
+"use client";
+import Button from '../button';
+import Input from '../input';
+import useZodForm from '../../../hooks/useZodForm';
+import { useState } from 'react';
+import { z } from 'zod';
+import { addProject, editProject } from '../../../actions/actions';
+import { useRouter } from 'next/navigation';
+import { type Project } from '@prisma/client';
+//import init from 'zod-empty';
+//import ErrorContext from '../../../state/error/error.context';
+//import ErrorProvider from '../../../state/error/error.provider';
+
+interface Props {
+  project?: Project;
+}
+
+export const schema = z.object({
+  name: z
+    .string()
+    .nonempty("Please enter project name"),
+  upworkId: z.string(),
+  hubstaffId: z.string(),
+  asanaId: z.string(),
+});
+
+export default function ProjectForm({ project }: Props) {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isValid, isSubmitting, isDirty },
+  } = useZodForm({
+    schema,
+    //defaultValues: init(schema),
```

Comment:
Comment this out - just pass `defaultValues` here instead of each `Input` separately